prov/efa/src/efa_env.c

```diff
@@ -67,7 +67,7 @@ struct efa_env efa_env = {
 	.efa_write_segment_size = 1073741824, /* need to confirm this constant. */
 	.rnr_retry = 3, /* Setting this value to EFA_RNR_INFINITE_RETRY makes the firmware retry indefinitey */
 	.host_id_file = "/sys/devices/virtual/dmi/id/board_asset_tag", /* Available on EC2 instances and containers */
-	.use_sm2 = false,
+	.intranode_provider = "shm",
```

Comment:
This change might make `FI_EFA_ENABLE_SHM_TRANSFER` ambiguous...

What about `FI_EFA_SHM_PROVIDER`?

Reply:
It is the selection for a peer provider and it doesn't have to be SHM... anything that implements the peer API (i.e.. tcp).

Reply:
Not sure about having 'peer' in the name either - peer api is an implementation detail to use another provider. 

My point is, if `FI_EFA_INTRANODE_PROVIDER=sm2`, then what should `FI_EFA_ENABLE_SHM_TRANSFER` control - does it control sm2?

Reply:
It seems we can deprecate the `FI_EFA_ENANLE_SHM_TRANSFER`, and replace it with setting `FI_EFA_INTRANODE_PROVIDER` to `efa`

Reply:
e.g.

FI_EFA_INTRANODE_PROVIDER = efa means use EFA device for intranode communication

Reply:
That seems reasonable to me. For the deprecation, can we also make `FI_EFA_ENANLE_SHM_TRANSFER=0` equivalent to `FI_EFA_INTRANODE_PROVIDER=efa` in the code? We need to update the doc, but I don't want to cause unexpected behaviors for customers who have been using this env

Reply:
Doesn't the ofi-nccl plugin use FI_EFA_ENABLE_SHM=0? We need to get them to update.

Reply:
done

Reply:
> Doesn't the ofi-nccl plugin use FI_EFA_ENABLE_SHM=0? We need to get them to update.

It's in the doc, but without this env it should also work.

Reply:
This change will make NCCL initialization faster b/c if we turn off SHM, we no longer initialize it now.

Reply:
NCCL disable shm by passing FI_REMOTE_COMM but not not FI_LOCAL_COMM. It did not use the env.